lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -566,18 +567,25 @@ def _get_meta_from_installed_dir(
         raise AnsibleError(
             'Collection tar file member {member!s} does not '
             'contain a valid json string.'.
-            format(member=n_manifest_json),
+            format(member=filename),
         )
-    else:
-        collection_info = manifest['collection_info']
+
+    return manifest
+
+
+def _get_meta_from_installed_dir(
+        b_path,  # type: bytes
+):  # type: (...) -> Dict[str, Optional[Union[str, List[str], Dict[str, str]]]]
+    manifest = _get_json_from_installed_dir(b_path, 'MANIFEST.json')
+    collection_info = manifest['collection_info']
 
     version = collection_info.get('version')
     if not version:
         raise AnsibleError(
-            u'Collection metadata file at `{meta_file!s}` is expected '
+            u'Collection metadata file `MANIFEST.json` at `{meta_file!s}` is expected '
```

Comment:
Could you keep `'MANIFEST.json'` in a var? I was hoping to extract https://github.com/ansible/ansible/blob/4a82e2c/lib/ansible/galaxy/dependency_resolution/dataclasses.py#L44 into a publicly exposed constant at some point...